### ISSUE TITLE
SPIR-V Shader optimization fails: Interfaces passed to OpEntryPoint must be of type OpTypeVariable

### ISSUE THREAD

**Issue by DasEtwas** [google/shaderc#1122]
```
$ glslc --version
2020.0
2020.2
8.13.3743

Target: SPIR-V 1.0
```

`$ glslc -g -x glsl --target-env=vulkan1.2 -O <file> -o <output_file>`

```glsl
#version 430

#extension GL_EXT_samplerless_texture_functions:enable

layout(constant_id = 1) const uint CACHE_ARRAY_SIZE = 1;
layout(constant_id = 2) const uint CACHE_ARRAY_COUNT = 1;

layout(set = 1, binding = 0) uniform sampler normals_sampler;
layout(set = 1, binding = 1) uniform sampler colors_sampler;
layout(set = 1, binding = 3) uniform sampler ground_tex_samples;

layout(set = 2, binding = 0) uniform texture2DArray normals[CACHE_ARRAY_COUNT];
layout(set = 2, binding = 1) uniform texture2DArray colors[CACHE_ARRAY_COUNT];
layout(set = 2, binding = 3) uniform texture2D ground_tex;

layout(location = 0) smooth in vec3 frag_uv_chunkDepth;
layout(location = 1) smooth in vec3 frag_normal;
layout(location = 2) smooth in vec3 frag_tangent;
layout(location = 3) flat in uint frag_slot;

// color
layout(location = 0) out vec4 diffuse_out;
// normals
layout(location = 1) out vec4 normal_out;

vec2 texcoords_for(texture2DArray texture) {
    float res = textureSize(texture, 0).x;
    float offset = 0.5 / res;            // Center of the first texel
    float range = 1.0 - (2.0 * offset);  // Distance between centers of first and last texels
    return frag_uv_chunkDepth.xy * range + offset;
}

void main() {
    vec3 base_normal_ = normalize(frag_normal);
    vec3 tangent_ = normalize(frag_tangent);
    vec3 bitangent = cross(base_normal_, tangent_);
    mat3 tangent_basis = mat3(tangent_, bitangent, base_normal_);

    uint array = frag_slot / CACHE_ARRAY_SIZE;
    uint layer = frag_slot % CACHE_ARRAY_SIZE;
    vec2 encoded = texture(sampler2DArray(normals[array], normals_sampler), vec3(texcoords_for(normals[array]), layer)).xy;
    vec3 decoded = vec3(encoded, sqrt(1 - dot(encoded.xy, encoded.xy)));
    vec3 normal = tangent_basis * decoded;

    vec4 base_color = texture(sampler2DArray(colors[array], colors_sampler), vec3(texcoords_for(colors[array]), layer));

    diffuse_out = vec4(texture(sampler2D(ground_tex, ground_tex_samples), frag_uv_chunkDepth.xy * pow(2.0, frag_uv_chunkDepth.z)).xyz * base_color.xyz, 1.0);
    normal_out = vec4(normal, 1.0);
}
```

error:
```
shaderc: internal error: compilation succeeded but failed to optimize: Interfaces passed to OpEntryPoint must be of type OpTypeVariable. Found OpFunctionParameter.
  %texture = OpFunctionParameter %_ptr_UniformConstant_8
```

**Comment by zoddicus** [comment]
@dneto0 can you take a look at this? I am unclear if this is a problem in glslc or glslang.

**Comment by dneto0** [comment]
I can reproduce this on v2020.0

**Comment by dneto0** [comment]
This was already fixed in release v2020.2
